### DISCUSSION TITLE
Scores not submitting on local build

### DISCUSSION THREAD

**Discussion by Ya-hwon** [ppy/osu#19747] [Q&A] [answered]
Hi,

I appear to not be able to submit scores on my local release build of lazer, no matter how long I wait - is this intended? I didn't find any reference to this behaviour.
I am logged in and have an active internet connection, I see other recent submissions on the beatmaps so that can't be it either, submission on default lazer works.

If this is intended - why?

**Comment by iSimpFor** [comment]
if i had to guess, it could be easily abused for rankings since you could just edit the code to your benefit

**Reply by Ya-hwon** [reply]
I don't think that's reasonable to assume because the server has no way of validating what kind of client it is talking to without the clients help - which can be changed clientside - this sort of lock wouldn't prevent anything (besides maybe the simplest attempts). Unless I'm missing something

**Comment by peppy** [comment, marked as answer]
Local debug builds use a different server. Are you saying they don't show there?

**Reply by Ya-hwon** [reply]
This is not a debug build (I run it with `dotnet run -c Release --project osu.Desktop`) - I already noticed that Debug builds use a different server as I was not logged in and not able to log in. But with the Release flag everything just appeared to work and use my default installation.

I do not see my scores in the leaderboards in the same app I played them in if that is what you mean. Otherwise I don't know how to validate this.

**Reply by nagi-desuuu** [reply]
Can you provide us with the version of your local build? We just want to make sure that you're running the latest commit.

**Reply by peppy** [reply]
That will not work. Please use binaries if you want to play online.

Eventually we should probably show a notification when score submission is not available.

**Reply by Ya-hwon** [reply]
Oh, ok.
Maybe this should be documented somewhere until it is added to the client.

May I ask why this is?

**Reply by peppy** [reply]
As mentioned, if anything it will be shown as a notification in the future.

As for why, it should be pretty obvious, but we cannot have users building the game with any local modifications and then playing online.

**Reply by Ya-hwon** [reply]
Sorry if I'm missing something obvious here but because of cheating then?

I would think that the server validates and recalculates the score of submitted plays anyway.
And it's not like this prevents cheating in general - maybe in its simplest form with the source available.[*]

My modification is very minuscule, I automatically restart a beatmap once it's over (so it loops until the loop is cancelled) - one little line for convenience's sake. This is the behaviour I want.
I did plan to make a pr for this but for that I would still need to figure out how to add and use checkbox settings (to make sure this is default off because I can't see many people wanting this) and even then I can't be sure that it would ever get accepted (I would understand if it didn't because as I said - unlikely to be popular and I'm not familiar with the policy on niche additions like this).

So until I got around to do try that I wanted to use it anyway. I feel like conceptually these little changes to the client are reasonable to want but adding everything everyone wants will probably result in a mess so local changes seem like the most sensible solution to me.

And unless the intention is to add limited user scriptability that maybe hook on ingame events (which sounds like a ton of work for something nearly no one would use) I don't see how this can otherwise be made possible.

[*] As I just found out it doesn't even do that, this is just a fence gate with no fence around it, changing the version hash to the one default lazer produces reenables score submission. The cheating prevention here being a `ListOfValidHashes.contains(request.VersionHash)` on the server? - That makes sense

But ok, I suppose my issue is actually solved with this so thank you for your time ^^
